lightning_examples/finetuning-scheduler/finetuning-scheduler.py

```diff
@@ -0,0 +1,712 @@
+            "model_name_or_path": model_name_or_path,
+            "task_name": self.task_name,
+            "experiment_id": self.experiment_id,
+        }
+        self.save_hyperparameters(self.init_hparams)
+        self.metric = datasets.load_metric("super_glue", self.task_name, experiment_id=self.experiment_id)
+        self.no_decay = ["bias", "LayerNorm.weight"]
+        self.finetuningscheduler_callback = None
+
+    def forward(self, **inputs):
+        return self.model(**inputs)
+
+    def training_step(self, batch, batch_idx):
+        outputs = self(**batch)
+        loss = outputs[0]
+        self.log("train_loss", loss, prog_bar=True)
+        return loss
+
+    def training_epoch_end(self, outputs: List[Any]) -> None:
+        if self.finetuningscheduler_callback:
+            self.log("finetuning_schedule_depth", float(self.finetuningscheduler_callback.curr_depth))
+
+    def validation_step(self, batch, batch_idx, dataloader_idx=0):
+        outputs = self(**batch)
+        val_loss, logits = outputs[:2]
+        if self.num_labels >= 1:
+            preds = torch.argmax(logits, axis=1)
+        elif self.num_labels == 1:
+            preds = logits.squeeze()
+        labels = batch["labels"]
+        self.log("val_loss", val_loss, prog_bar=True)
+        metric_dict = self.metric.compute(predictions=preds, references=labels)
+        self.log_dict(metric_dict, prog_bar=True)
+
+    def _init_param_groups(self) -> List[Dict]:
+        """Initialize the parameter groups. Used to ensure weight_decay is not applied to our specified bias
+        parameters when we initialize the optimizer.
+
+        Returns:
+            List[Dict]: A list of parameter group dictionaries.
+        """
+        return [
+            {
+                "params": [
+                    p
+                    for n, p in self.model.named_parameters()
+                    if not any(nd in n for nd in self.no_decay) and p.requires_grad
+                ],
+                "weight_decay": self.optimizer_init["weight_decay"],
+            },
+            {
+                "params": [
+                    p
+                    for n, p in self.model.named_parameters()
+                    if any(nd in n for nd in self.no_decay) and p.requires_grad
+                ],
+                "weight_decay": 0.0,
+            },
+        ]
+
+    def configure_optimizers(self):
+        # the phase 0 parameters will have been set to require gradients during setup
+        # you can initialize the optimizer with a simple requires.grad filter as is often done,
+        # but in this case we pass a list of parameter groups to ensure weight_decay is
+        # not applied to the bias parameter (for completeness, in this case it won't make much
+        # performance difference)
+        optimizer = AdamW(params=self._init_param_groups(), **self.optimizer_init)
+        scheduler = {"scheduler": CosineAnnealingWarmRestarts(optimizer, **self.lr_scheduler_init)}
```

Comment:
```suggestion
        scheduler = {"scheduler": CosineAnnealingWarmRestarts(optimizer, **self.lr_scheduler_init), "interval": "step"}
```
if this is intended to be called with each optimizer step, you need to define the interval